src/waveform/renderers/waveformmark.cpp

```diff
@@ -5,21 +5,23 @@
 
 #include "waveformmark.h"
 
-WaveformMark::WaveformMark(int hotCue)
+WaveformMark::WaveformMark( const QString& group,
+                            const QDomNode& node,
+                            const SkinContext& context,
+                            const WaveformSignalColors& signalColors,
+                            int hotCue,
+                            QString item)
     : m_iHotCue(hotCue) {
-}
-
-void WaveformMark::reset(int hotCue) {
-    WaveformMark(hotCue).swap(*this);
-}
-
-void WaveformMark::setup(const QString& group, const QDomNode& node,
-                         const SkinContext& context,
-                         const WaveformSignalColors& signalColors) {
-    QString item = context.selectString(node, "Control");
+    if(item.isEmpty())
+        item = context.selectString(node, "Control");
```

Comment:
The idea was to calculate "item" always before the constructor call. 
But reading the code again, it looks good now. 
I hope I find time to test and merge it soon.